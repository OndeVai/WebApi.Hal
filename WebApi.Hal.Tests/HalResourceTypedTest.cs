#region

using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net.Http;
using ApprovalTests;
using ApprovalTests.Reporters;
using WebApi.Hal.Tests.Representations;
using Xunit;

#endregion

namespace WebApi.Hal.Tests
{
    public class HalResourceTypedTest
    {
        readonly OrganisationRepresentationTyped resource;

        public HalResourceTypedTest()
        {
            var model = new Organisation(1, "Org Name");
            var emps = new List<Employee>
            {
                new Employee(1, "emp1", 1),
                new Employee(11, "emp2", 2),
                new Employee(12, "emp1", 4),
            };

            var employeeRepresentations = emps.Select(e => new EmployeeRepresentation(e)).ToList();
            model.Employees = new EmployeeRepresentationList(employeeRepresentations);
            resource = new OrganisationRepresentationTyped(model);
        }

        [Fact]
        [UseReporter(typeof (DiffReporter))]
        public void organisation_get_json_test()
        {
            // arrange
            var mediaFormatter = new JsonHalMediaTypeFormatter {Indent = true};
            var content = new StringContent(string.Empty);
            var type = resource.GetType();

            // act
            using (var stream = new MemoryStream())
            {
                mediaFormatter.WriteToStreamAsync(type, resource, stream, content, null);
                stream.Seek(0, SeekOrigin.Begin);
                var serialisedResult = new StreamReader(stream).ReadToEnd();

                // assert
                Approvals.Verify(serialisedResult);
            }
        }

        [Fact]
        [UseReporter(typeof (DiffReporter))]
        public void organisation_get_xml_test()
        {
            // arrange
            var mediaFormatter = new XmlHalMediaTypeFormatter();
            var content = new StringContent(string.Empty);
            var type = resource.GetType();

            // act
            using (var stream = new MemoryStream())
            {
                mediaFormatter.WriteToStreamAsync(type, resource, stream, content, null);
                stream.Seek(0, SeekOrigin.Begin);
                var serialisedResult = new StreamReader(stream).ReadToEnd();

                // assert
                Approvals.Verify(serialisedResult);
            }
        }
    }
}