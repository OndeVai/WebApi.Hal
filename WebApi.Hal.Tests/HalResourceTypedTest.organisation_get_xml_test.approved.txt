<?xml version="1.0" encoding="utf-8"?>
<resource rel="organisation" href="/api/organisations/1">
  <link rel="related" href="/api/other" />
  <Id>1</Id>
  <Name>Org Name</Name>
  <resource rel="employees" href="/api/employees" name="Employees">
    <resource rel="" href="/api/employees/1">
      <Id>1</Id>
      <Title>emp1</Title>
      <Rank>1</Rank>
    </resource>
    <resource rel="" href="/api/employees/11">
      <Id>11</Id>
      <Title>emp2</Title>
      <Rank>2</Rank>
    </resource>
    <resource rel="" href="/api/employees/12">
      <Id>12</Id>
      <Title>emp1</Title>
      <Rank>4</Rank>
    </resource>
  </resource>
</resource>